### PR TITLE
refactor(log): use LOG_ERROR_F instead of LOG_ERROR (2/2)

### PR DESCRIPTION
**Pull request by WHBANG**
### What problem does this PR solve? 
https://github.com/apache/incubator-pegasus/issues/1305

This patch aim to use LOG_ERROR_F instead of LOG_ERROR for all the remain parts:

- Remove the defination of LOG_ERROR
- Use LOG_ERROR_F instead of LOG_ERROR
- Use LOG_ERROR_PREFIX instead of LOG_ERROR for class pegasus_server_impl
- Some format fixes

### CODE REVIEW

File: src/client/partition_resolver_simple.cpp [hunk 1]
```diff
@@ -311,28 +311,28 @@ void partition_resolver_simple::query_config_reply(error_code err,
                 }
             }
         } else if (resp.err == ERR_OBJECT_NOT_FOUND) {
-            LOG_ERROR("%s.client: query config reply, gpid = %d.%d, err = %s",
-                      _app_name.c_str(),
-                      _app_id,
-                      partition_index,
-                      resp.err.to_string());
+            LOG_ERROR_F("{}.client: query config reply, gpid = {}.{}, err = {}",
```
Comment (acelyc111) [inline src/client/partition_resolver_simple.cpp:314]: You can use `LOG_ERROR_PREFIX` macro instead.

File: src/failure_detector/failure_detector.cpp [hunk 2]
```diff
@@ -227,13 +227,13 @@ void failure_detector::check_all_records()
...
                 now + _check_interval_milliseconds - record.last_send_time_for_beacon_with_ack >
                     _lease_milliseconds) {
-                LOG_ERROR("master %s disconnected, now=%" PRId64 ", last_send_time=%" PRId64
-                          ", now+check_interval-last_send_time=%" PRId64,
-                          record.node.to_string(),
-                          now,
-                          record.last_send_time_for_beacon_with_ack,
-                          now + _check_interval_milliseconds -
-                              record.last_send_time_for_beacon_with_ack);
+                LOG_ERROR_F("master {} disconnected, now={:#018x}, last_send_time={:#018x}, "
```
Comment (acelyc111) [inline src/failure_detector/failure_detector.cpp:230]: is it necessary to use `:#018x`?

File: src/meta/meta_backup_service.cpp [hunk 3]
```diff
@@ -625,10 +625,10 @@ void policy_context::sync_backup_to_remote_storage_unlocked(const backup_info &b
                 LOG_WARNING_F("{}: empty callback", _policy.policy_name);
             }
         } else if (ERR_TIMEOUT == err) {
-            LOG_ERROR("%s: sync backup info(" PRId64
-                      ") to remote storage got timeout, retry it later",
-                      _policy.policy_name.c_str(),
-                      b_info.backup_id);
+            LOG_ERROR_F(
+                "{}: sync backup info({:#018x}) to remote storage got timeout, retry it later",
```
Comment (acelyc111) [inline src/meta/meta_backup_service.cpp:629]: why use `:#018x` ?

File: src/nfs/nfs_server_impl.cpp [hunk 4]
```diff
@@ -208,9 +207,9 @@ void nfs_service_impl::on_get_file_size(
 
             struct stat st;
             if (0 != ::stat(file_path.c_str(), &st)) {
-                LOG_ERROR("{nfs_service} get stat of file %s failed, err = %s",
-                          file_path.c_str(),
-                          strerror(errno));
+                LOG_ERROR_F("[nfs_service] get stat of file {} failed, err = {}",
+                            file_path,
+                            strerror(errno));
```
Comment (acelyc111) [inline src/nfs/nfs_server_impl.cpp:212]: it's recommend to use `dsn::utils::safe_strerror` instead.

File: src/runtime/rpc/asio_net_provider.cpp [hunk 5]
```diff
@@ -148,7 +148,7 @@ void asio_network_provider::do_accept()
         if (!ec) {
             auto remote = socket->remote_endpoint(ec);
             if (ec) {
-                LOG_ERROR("failed to get the remote endpoint: %s", ec.message().data());
+                LOG_ERROR_F("failed to get the remote endpoint: {}", ec.message().data());
```
Comment (acelyc111) [inline src/runtime/rpc/asio_net_provider.cpp:151]: ec.message() is ok?

File: src/client/partition_resolver_simple.cpp (hunk 1 above)
Comment (WHBANG) [reply src/client/partition_resolver_simple.cpp]: done

File: src/failure_detector/failure_detector.cpp (hunk 2 above)
Comment (WHBANG) [reply src/failure_detector/failure_detector.cpp]: fixed

File: src/meta/meta_backup_service.cpp (hunk 3 above)
Comment (WHBANG) [reply src/meta/meta_backup_service.cpp]: fixed

File: src/nfs/nfs_server_impl.cpp (hunk 4 above)
Comment (WHBANG) [reply src/nfs/nfs_server_impl.cpp]: done

File: src/runtime/rpc/asio_net_provider.cpp (hunk 5 above)
Comment (WHBANG) [reply src/runtime/rpc/asio_net_provider.cpp]: done

File: src/client/partition_resolver_simple.cpp [hunk 6]
```diff
@@ -311,28 +311,18 @@ void partition_resolver_simple::query_config_reply(error_code err,
...
 
             client_err = resp.err;
         }
     } else {
-        LOG_ERROR("%s.client: query config reply, gpid = %d.%d, err = %s",
-                  _app_name.c_str(),
-                  _app_id,
-                  partition_index,
-                  err.to_string());
+        LOG_ERROR_F("query config reply, gpid = {}.{}, err = {}", _app_id, partition_index, err);
```
[the remark this answers is not in the document]
Comment (WHBANG) [reply src/client/partition_resolver_simple.cpp]: done

File: src/meta/meta_backup_service.cpp [hunk 7]
```diff
@@ -1363,10 +1357,10 @@ void backup_service::do_update_policy_to_remote_storage(
                            p.policy_name);
                 p_context_ptr->set_policy(p);
             } else if (err == ERR_TIMEOUT) {
-                LOG_ERROR("update backup policy to remote storage failed, policy_name = %s, retry "
-                          "after %" PRId64 "(ms)",
-                          p.policy_name.c_str(),
-                          _opt.meta_retry_delay_ms.count());
+                LOG_ERROR_F("update backup policy to remote storage failed, policy_name = {}, "
+                            "retry after {:#018x} (ms)",
```
[the remark this answers is not in the document]
Comment (WHBANG) [reply src/meta/meta_backup_service.cpp]: done

File: src/meta/meta_backup_service.cpp [hunk 8]
```diff
@@ -1329,9 +1324,8 @@ void backup_service::do_add_policy(dsn::message_ex *req,
                 }
                 p->start();
             } else if (err == ERR_TIMEOUT) {
-                LOG_ERROR("create backup policy on remote storage timeout, retry after %" PRId64
-                          "(ms)",
-                          _opt.meta_retry_delay_ms.count());
+                LOG_ERROR_F("create backup policy on remote storage timeout, retry after {} (ms)",
```
[the remark this answers is not in the document]
Comment (WHBANG) [reply src/meta/meta_backup_service.cpp]: done

File: src/server/pegasus_server_impl.cpp [hunk 9]
```diff
@@ -365,11 +359,9 @@ void pegasus_server_impl::on_multi_get(multi_get_rpc rpc)
     }
 
     if (!is_filter_type_supported(request.sort_key_filter_type)) {
-        LOG_ERROR("%s: invalid argument for multi_get from %s: "
-                  "sort key filter type %d not supported",
-                  replica_name(),
-                  rpc.remote_address().to_string(),
-                  request.sort_key_filter_type);
+        LOG_ERROR_F("invalid argument for multi_get from {}: sort key filter type {} not supported",
```
[the remark this answers is not in the document]
Comment (WHBANG) [reply src/server/pegasus_server_impl.cpp]: done

**Review by empiredan — APPROVED** [review]

File: src/meta/meta_backup_service.cpp [hunk 10]
```diff
@@ -1363,10 +1357,10 @@ void backup_service::do_update_policy_to_remote_storage(
                            p.policy_name);
                 p_context_ptr->set_policy(p);
             } else if (err == ERR_TIMEOUT) {
-                LOG_ERROR("update backup policy to remote storage failed, policy_name = %s, retry "
-                          "after %" PRId64 "(ms)",
-                          p.policy_name.c_str(),
-                          _opt.meta_retry_delay_ms.count());
+                LOG_ERROR_F("update backup policy to remote storage failed, policy_name = {}, "
+                            "retry after {:#018x} ms",
```
Comment (acelyc111) [inline src/meta/meta_backup_service.cpp:1361]:
```suggestion
                            "retry after {} ms",
```

**Review by acelyc111 — APPROVED** [review]
Good job!

File: src/meta/meta_backup_service.cpp (hunk 10 above)
Comment (WHBANG) [reply src/meta/meta_backup_service.cpp]: fixed

**Review by empiredan — APPROVED** [review]

**Pull request merged**